File: 이코테 복습(python, c++, java)/구현/python/상하좌우.py
N = int(input())
x, y = 1, 1
plans = input().split()

dx = [0, 0, -1, 1]
dy = [-1, 1, 0, 0]
move = ['L', 'R', 'U', 'D']

for p in plans:
    for i in range(len(move)):
        if p == move[i]:
            nx = x + dx[i]
            ny = y + dy[i]
            break

    if nx < 1 or ny < 1 or nx > N or ny > N:
        continue
    x, y = nx, ny
print(x, y)

File: 이코테 복습(python, c++, java)/구현/java/상하좌우.java
package javastudy.이코테.구현;

import java.util.Scanner;

public class 상하좌우 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        sc.nextLine(); //버퍼 비우기
        String[] plans = sc.nextLine().split(" ");
        int x = 1, y = 1;

        int[] dx = {0, 0, -1, 1};
        int[] dy = {-1, 1, 0, 0};
        char[] moveTypes = {'L', 'R', 'U', 'D'};

        for (int i = 0; i < plans.length; i++) {
            char p = plans[i].charAt(0);
            int nx = -1, ny = -1;

            for (int j = 0; j < 4; j++) {
                if (p == moveTypes[j]) {
                    nx = x + dx[j];
                    ny = y + dy[j];
                    break;
                }
            }

            if (nx < 1 || ny < 1 || nx > n || ny > n) {
                continue;
            }

            x = nx;
            y = ny;
        }
        System.out.println(x + " " + y);
    }
}
